h2o-core/src/main/java/hex/Model.java

```diff
@@ -281,6 +281,16 @@ public void setTrain(Key<Frame> train) {
       CategoricalEncodingScheme(boolean needResponse) { _needResponse = needResponse; }
       final boolean _needResponse;
       boolean needsResponse() { return _needResponse; }
+      public static CategoricalEncodingScheme fromGenModel(CategoricalEncoding encoding) {
+        if (encoding == null)
+          return null;
+        final String name = encoding.name();
+        for (CategoricalEncodingScheme scheme : CategoricalEncodingScheme.values()) {
+          if (name.equals(scheme.name())) 
+            return scheme;
+        }
+        throw new UnsupportedOperationException("Unknown encoding " + encoding);
```

Comment:
```suggestion
        try {
            return Enum.valueOf(CategoricalEncodingScheme.class, encoding.name());
        } catch (IllegalArgumentException iae) {
            throw new UnsupportedOperationException("Unknown encoding " + encoding);
        }
```
You may not even need to translate the `IllegalArgumentException` given that `encoding.name()` already comes from an enum.
Which btw means that it could have been implemented differently: I would have made `CategoricalEncoding` an interface, and add a `getInstace()` or `newInstance()` method to the `CategoricalEncodingScheme` enum.
This way, no need to translate between 2 enums and the separation of responsibility is made clear between the scheme and the implementation.